modules/3d/include/opencv2/3d.hpp

```diff
@@ -468,77 +468,308 @@ can be found in:
+            ok = ok && (this->stepNormInf == other.stepNormInf);
+            ok = ok && (this->checkRelEnergyChange == other.checkRelEnergyChange);
+            ok = ok && (this->checkMinGradient == other.checkMinGradient);
+            ok = ok && (this->checkStepNorm == other.checkStepNorm);
+
+            ok = ok && (this->geodesic == other.geodesic);
+            ok = ok && abs(this->hGeo - other.hGeo) < eps;
+            ok = ok && abs(this->geoScale - other.geoScale) < eps;
+
+            ok = ok && abs(this->stepNormTolerance - other.stepNormTolerance) < eps;
+            ok = ok && abs(this->relEnergyDeltaTolerance - other.relEnergyDeltaTolerance) < eps;
+            ok = ok && abs(this->minGradientTolerance - other.minGradientTolerance) < eps;
+            ok = ok && abs(this->smallEnergyTolerance - other.smallEnergyTolerance) < eps;
+
+            ok = ok && (this->maxIterations == other.maxIterations);
+
+            ok = ok && abs(this->initialLambda - other.initialLambda) < eps;
+            ok = ok && abs(this->initialLmUpFactor - other.initialLmUpFactor) < eps;
+            ok = ok && abs(this->initialLmDownFactor - other.initialLmDownFactor) < eps;
+
+            return ok;
+        }
+
+        Settings& jacobiScalingS          (bool         v) { jacobiScaling           = v; return *this; }
+        Settings& upDoubleS               (bool         v) { upDouble                = v; return *this; }
+        Settings& useStepQualityS         (bool         v) { useStepQuality          = v; return *this; }
+        Settings& clampDiagonalS          (bool         v) { clampDiagonal           = v; return *this; }
+        Settings& stepNormInfS            (bool         v) { stepNormInf             = v; return *this; }
+        Settings& checkRelEnergyChangeS   (bool         v) { checkRelEnergyChange    = v; return *this; }
+        Settings& checkMinGradientS       (bool         v) { checkMinGradient        = v; return *this; }
+        Settings& checkStepNormS          (bool         v) { checkStepNorm           = v; return *this; }
+        Settings& geodesicS               (bool         v) { geodesic                = v; return *this; }
+        Settings& hGeoS                   (double       v) { hGeo                    = v; return *this; }
+        Settings& geoScaleS               (double       v) { geoScale                = v; return *this; }
+        Settings& stepNormToleranceS      (double       v) { stepNormTolerance       = v; return *this; }
+        Settings& relEnergyDeltaToleranceS(double       v) { relEnergyDeltaTolerance = v; return *this; }
+        Settings& minGradientToleranceS   (double       v) { minGradientTolerance    = v; return *this; }
+        Settings& smallEnergyToleranceS   (double       v) { smallEnergyTolerance    = v; return *this; }
+        Settings& maxIterationsS          (unsigned int v) { maxIterations           = v; return *this; }
```

Comment:
> unsigned int

This type may be non-friendly with bindings.

Reply:
Replaced by int